README.md

```diff
@@ -38,6 +43,58 @@ If you need support, please either raise an issue on Github or ask on [Slack](ht
 
 Substra warmly welcomes any contribution. Feel free to fork the repo and create a pull request.
 
+## How to test
+
+You need to install `substrafl`, `substra` and `substra-tools` locally. 
+Clone this repo, then at the top level run in your virtual env:
+
+```sh
+pip install -e ".[dev]"
+``
+
+Clone [`substra`](https://github.com/Substra/substra) locally, go to the top level directory of `substra` and run (still in your virtual env):
+
+```sh
+pip install -e ".[dev]"
+```
+
+Then clone [`substra-tools`](https://github.com/Substra/substra-tools) locally, go to the top level directory of `substra-tools` and run (still in your virtual env):
+
+```sh
+pip install -e ".[test]"
+```
+
+Now you can use the following command from `subtrafl` top level directory to run tests:
+
+```sh
+make test-subprocess
+```
+
+### Running advanced test suites
+
+Substra can be used in three different modes: using subprocesses, using Docker and using Kubernetes. 
+
+The command `make test-subprocess` runs the test suite in subprocess mode. It's lightweight and perfect to start. 
+
+If you want to test with the Docker mode, you will need Docker installed and running on your machine. 
+If necessary, you can install it using [Docker Desktop](https://www.docker.com/products/docker-desktop/). 
+
+Then you can run the test suites in subprocess and Docker mode: 
+
+```sh 
+make test-local
+``
+
+Please be warned that some of these tests are slow and the whole test suite might require a couple hours to complete. 
+
+If you want to try out a local deployment with Kubernetes, please follow the [installation instructions](https://docs.substra.org/en/stable/contributing/getting-started.html) provided in the doc. 
+Then you should be able to run the remote tests:
+
+```sh
+make test-remote
+```
+
+
 
 # Appendix
```

Comment:
```suggestion
```
We can delete this as there is no appendix right now

Reply:
There is actually the instructions for building the documentation in the appendix